docs/guides/upgrade-apps-without-downtime.md

```diff
@@ -0,0 +1,182 @@
+  and then reloads NGINX. After that, NGINX will start proxying traffic to that Endpoint.
+- If an Endpoint is removed, NGINX Kubernetes Gateway removes the corresponding upstream server from NGINX. After
+  a reload, NGINX will stop proxying traffic to it. However, it will finish proxying any pending requests to that
+  server,
+  which prevents downtime.
+
+As a result, as long you have more than one ready Endpoint, the clients should not experience any downtime during
+an upgrade.
+
+> It is a good practice to configure a [Readiness probe][readiness-probe] in the Deployment so that a pod can advertise
+> when it is ready to receive traffic. Note that NGINX Kubernetes Gateway will not add any Endpoint to NGINX that is not
+> ready.
+
+[readiness-probe]:https://kubernetes.io/docs/tasks/configure-pod-container/configure-liveness-readiness-startup-probes/
+
+The upgrade methods from the next sections will change upstream servers in NGINX.
+
+## Rolling Deployment Upgrade
+
+To start a [rolling Deployment upgrade][rolling-upgrade], you update the Deployment to use the new version tag of
+the application. As a result, Kubernetes terminates the Pods with the old version and create new ones. By default,
+Kubernetes also ensures that some number of pods always stays available during the upgrade.
+
+[rolling-upgrade]:https://kubernetes.io/docs/concepts/workloads/controllers/deployment/#rolling-update-deployment
+
+Such an upgrade will add new upstream servers to NGINX and remove the old ones. As long as the number
+of Pods (ready Endpoints) during an upgrade does not reach zero, NGINX will be able to proxy traffic, and thus prevent
+any downtime.
+
+To use this method, it is not necessary to update the HTTPRoute.
+
+## Blue-Green Deployments
+
+With this method, you deploy a new version of the application (blue version) as a separate Deployment,
+while the old version (green) keeps running and handling client traffic. Next, you switch the traffic from the
+green version to the blue. If the blue works as expected, you terminate the green or switch the traffic back to it
+otherwise.
+
+There are two ways to switch the traffic:
+
+- Update the Service selector to select the Pods of the blue version instead of the green. As a result, NGINX Kubernetes
+  Gateway removes the green upstream servers from NGINX and add the blue ones. With this approach, it is not
+  necessary to update the HTTPRoute.
+- Create a separate Service for the blue version and update the backend reference in the HTTPRoute to reference this
+  Service, which leads to the same result as with the previous option.
+
+## Canary Releases
+
+To support canary releases, you can implement an approach with two Deployments behind the same Service (see
+[Canary deployment][canary] in the Kubernetes documentation). For example, if you have an old Deployment with four Pods
+and a new Deployment with one Pod, NGINX will proxy about 80% of client requests to the old Pods and 20% to the new Pod.
```

Comment:
Is this true given that the lb algorithm we use is `random two least_conn`?

Reply:
good question!

how about this version:

To support canary releases, you can implement an approach with two Deployments behind the same Service (see
[Canary deployment]() in the Kubernetes documentation). However, this approach lacks precision for defining the traffic split between the old and the canary version. You can greatly influence it by controlling the number of pods (for example, four Pods of the old version and one Pod of the canary). However, note that NGINX Kubernetes Gateway uses [`random two least_conn`](https://nginx.org/en/docs/http/ngx_http_upstream_module.html#random) load balancing method, which doesn't guarantee an 80/20 split based on the number of pods in the given example.

Reply:
LGTM!